scheduler/reconcile.go

```diff
@@ -844,16 +844,29 @@ func (a *allocReconciler) handleDelayedReschedules(rescheduleLater []*delayedRes
 	})
 
 	var evals []*structs.Evaluation
-	var allocIDs []string
 	nextReschedTime := rescheduleLater[0].rescheduleTime
 	allocIDToFollowupEvalID := make(map[string]string, len(rescheduleLater))
+	// Create a new eval for the first batch
```

Comment:
Can we return early if `len(rescheduleLater) == 0`. This would panic if passed in an empty list right now